### PR TITLE
aws-assume-role: init at 0.3.2

### PR DESCRIPTION
**Pull request by averyvigolo**
###### Description of changes

assume-role is an AWS CLI tool which makes it easy to assume a different AWS role in your terminal session.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by KAction** [comment]
I think adding yourself to maintainer list should be separate commit.

````
7.1.8. maintainers
A list of the maintainers of this Nix expression. Maintainers are defined in nixpkgs/maintainers/maintainer-list.nix. There is no restriction to becoming a maintainer, just add yourself to that list in a separate commit titled “maintainers: add alice”, and reference maintainers with maintainers = with lib.maintainers; [ alice bob ].
```
https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers

**Review by KAction — APPROVED** [review]

**Comment by williamvds** [comment]
Thanks for the review @KAction!

I have run this on a work machine with the AWS CLI all configured, it works for me.

I've split the maintainer addition to a separate commit, and have installed the readme under docs. I assume these will have to be sought out manually, as it hasn't been installed as a man page?

**Comment by KAction** [comment]
Yes, if it is not man, user will have to do `ls ~/.nix-profile/share/doc` or equivalent.

Please install into `$out/share/doc/$name` instead of `$out/share/doc/$pname`. This way documentation of different versions is co-installable. Not that it is very relevant for this package, but for consistency. Also please add `outputs = [ "out" "doc" ]`.

**Comment by williamvds** [comment]
Thanks, I've amended the install path and added the `outputs`

**Review by KAction — APPROVED** [review]

**Comment by NickCao** [comment]
This package seems to be abandoned by upstream, and there is builtin support for assuming roles in awscli: https://awscli.amazonaws.com/v2/documentation/api/latest/reference/sts/assume-role.html

**Comment by williamvds** [comment]
> This package seems to be abandoned by upstream, and there is builtin support for assuming roles in awscli: https://awscli.amazonaws.com/v2/documentation/api/latest/reference/sts/assume-role.html

Yes, the AWS CLI can fulfill the step of calling the API to request a token. assume-role provides some extra conveniences, such as:

* parsing your config files so you can use a human-readable profile name
* requesting and using a multi-factor authentication code, if necessary
* setting the appropriate environment variables with the resulting access key etc

**Review by NickCao — APPROVED** [review]
not sure if it's really necessary to split doc into a separate output, overall LGTM.

File: pkgs/tools/admin/aws-assume-role/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ pkgs
+, lib
+, fetchFromGitHub
+, fetchurl
```
Comment (lukegb) [inline pkgs/tools/admin/aws-assume-role/default.nix:4]: This seems unused?

**Review by lukegb — CHANGES_REQUESTED** [review]

**Review by lukegb — APPROVED** [review]

**Pull request merged**